### PR TITLE
debug-tools: Binlog event blackhole

### PR DESCRIPTION
**Pull request by csuzhangxc**
### What problem does this PR solve? 

add a _binlog-event-blackhole_ tool to test the speed of fetching binlog event.

### What is changed and how it works?

add a _binlog-event-blackhole_ tool.

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
	- run _binlog-event-blackhole_ tool with different args

for more information, see _README.md_ under the _binlog-event-blackhole_ folder.

### CODE REVIEW

File: debug-tools/binlog-event-blackhole/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  -u string
+    	master's username
+```
+
+## Result
+
+When exiting, the result will be output to the log as following
+
+```log
+[2019/08/12 15:44:19.269 +08:00] [INFO] [main.go:95] ["binlog-event-blackhole exit"] [event-count=35] [byte-count=2360] [duration=705.627314ms] [tps=49.601254522865595] ["speed (byte/s)"=3344.541733541794]
```
Comment (GregoryIan) [inline debug-tools/binlog-event-blackhole/README.md]:
```suggestion
[2019/08/12 15:44:19.269 +08:00] [INFO] [main.go:95] ["binlog-event-blackhole exit"] [event-count=35] [byte-count=2360] [duration=705.627314ms] [tps=49.601254522865595] ["throughput (byte/s)"=3344.541733541794]
```

File: debug-tools/binlog-event-blackhole/README.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+```
+
+| Item | Description |
+|:------|:---- |
+| event-count | The total events have received from the upstream master |
+| byte-count | The total bytes have received from the upstream master |
+| duration | The duration has be taken to fetch binlog events |
+| tps | The events have received per second |
+| speed | The speed of fetching binlog event data (bytes/second) |
```
Comment (GregoryIan) [inline debug-tools/binlog-event-blackhole/README.md]:
```suggestion
| throughput | The throughput of fetching binlog event data (bytes/second) |
```

File: debug-tools/binlog-event-blackhole/fetcher.go [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+	err := conn.SetReadDeadline(deadline)
+	if err != nil {
+		return errors.Annotatef(err, "set connection read deadline to %v", deadline)
+	}
+
+	return errors.Trace(conn.Close())
+}
+
+// readEventsWithGoMySQL reads binlog events from the master server with `go-mysql` pkg.
+func readEventsWithGoMySQL(ctx context.Context, conn *client.Conn) (uint64, uint64, time.Duration, error) {
```
Comment (GregoryIan) [inline debug-tools/binlog-event-blackhole/fetcher.go]: the differences with `readEventsWithoutGoMySQL` is  the `"github.com/siddontang/go-mysql/client".Conn` use a buffer, I originally thought that it also contains more logic, such as decoding/parsing.  

Let's do this now, you can add the complete `go-mysql` logic later.

File: debug-tools/binlog-event-blackhole/main.go [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+	}
+	if err != nil {
+		log.L().Error("read events", zap.Error(err))
+	}
+
+	tps := float64(eventCount) / duration.Seconds()
+	speed := float64(byteCount) / duration.Seconds()
+	log.L().Info("binlog-event-blackhole exit",
+		zap.Uint64("event-count", eventCount), zap.Uint64("byte-count", byteCount),
+		zap.Duration("duration", duration), zap.Float64("tps", tps), zap.Float64("speed (byte/s)", speed))
```
Comment (GregoryIan) [inline debug-tools/binlog-event-blackhole/main.go]:
```suggestion
		zap.Duration("duration", duration), zap.Float64("tps", tps), zap.Float64("throughput (byte/s)", speed))
```

**Comment by GregoryIan** [comment]
Rest LGTM

File: debug-tools/binlog-event-blackhole/README.md (hunk 1 above)
Comment (csuzhangxc) [reply debug-tools/binlog-event-blackhole/README.md]: addressed in https://github.com/pingcap/dm/pull/235/commits/50c2b563b27b719c3458d5ddbafd95d27595bbed

File: debug-tools/binlog-event-blackhole/README.md (hunk 2 above)
Comment (csuzhangxc) [reply debug-tools/binlog-event-blackhole/README.md]: addressed in https://github.com/pingcap/dm/pull/235/commits/50c2b563b27b719c3458d5ddbafd95d27595bbed

File: debug-tools/binlog-event-blackhole/main.go (hunk 4 above)
Comment (csuzhangxc) [reply debug-tools/binlog-event-blackhole/main.go]: addressed in https://github.com/pingcap/dm/pull/235/commits/50c2b563b27b719c3458d5ddbafd95d27595bbed.

File: debug-tools/binlog-event-blackhole/fetcher.go (hunk 3 above)
Comment (csuzhangxc) [reply debug-tools/binlog-event-blackhole/fetcher.go]: Yeh. after doing some benchmark later, I think more (fine-grained) things we can do.

**Pull request merged**